usr/initiator_common.c

```diff
@@ -273,12 +273,16 @@ static int host_set_param(struct iscsi_transport *t,
 
 static void print_param_value(enum iscsi_param param, void *value, int type)
 {
-	log_debug(3, "set operational parameter %d to:", param);
+	char *buff = NULL;
 
-	if (type == ISCSI_STRING)
-		log_debug(3, "%s", value ? (char *)value : "NULL");
-	else
-		log_debug(3, "%u", *(uint32_t *)value);
+	if (type == ISCSI_STRING) {
+		buff = value ? (char *)value : "NULL";
+	} else {
+		buff = malloc(12);
```

Comment:
is this malloced memory being leaked?

Reply:
For such a small buffer, I think I'd rather see this on the stack to avoid needing to free it manually.

Or just put two complete log_debug calls in the two branches and do away with the snprintf and buff.
```
if (type == ISCSI_STRING)
    log_debug(3, "set operational parameter %d to %s", param, value ? (char *)value : "NULL");
else
    log_debug(3, "set operational parameter %d to %u", param, *(uint32_t *)value);
```

This looks like a victim of removing the manual newlines and making each log call append them.

Reply:
> For such a small buffer, I think I'd rather see this on the stack to avoid needing to free it manually.
> 
> Or just put two complete log_debug calls in the two branches and do away with the snprintf and buff.
> 
> ```
> if (type == ISCSI_STRING)
>     log_debug(3, "set operational parameter %d to %s", param, value ? (char *)value : "NULL");
> else
>     log_debug(3, "set operational parameter %d to %u", param, *(uint32_t *)value);
> ```
> 
> This looks like a victim of removing the manual newlines and making each log call append them.

done